jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -433,6 +434,12 @@ def _image_viewer_update(self, viewer, x, y):
             # use WCS to expose the wavelength for a 2d spectrum shown in pixel space
             try:
                 wave, pixel = image.coords.pixel_to_world(x, y)
+                if wave is not None:
+                    equivalencies = all_flux_unit_conversion_equivs(cube_wave=wave)
+                    wave = wave.to(self.app._get_display_unit('spectral'),
+                                   equivalencies=equivalencies)
+                    self._dict['spectral_axis'] = wave.value
+                    self._dict['spectral_axis:unit'] = wave.unit.to_string()
```

Comment:
where were these handled before (I don't see them in the else 🤔)?

Reply:
X-axis conversions were not previously handled in Specviz2d, so they were previously always using the native units and value when mousing over